src/main/java/org/opensearch/neuralsearch/processor/combination/BaseScoreCombinationFactory.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.processor.combination;
+
+import java.util.List;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.extern.log4j.Log4j2;
+import org.apache.lucene.search.Sort;
+import org.opensearch.common.Nullable;
+import org.opensearch.neuralsearch.processor.CompoundTopDocs;
+import org.opensearch.search.query.QuerySearchResult;
+
+/**
+ * Base class for creating score combination DTO.
+ */
+@Log4j2
+public abstract class BaseScoreCombinationFactory {
```

Comment:
Why do we need this class? We can just have CombineScoresDTO as a class, without wrapping abstract class BaseScoreCombinationFactory